lib/web_ui/lib/ui_web/src/ui_web/navigation/url_strategy.dart

```diff
@@ -0,0 +1,155 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:ui/src/engine.dart';
+import 'package:ui/ui.dart' as ui;
+
+import 'platform_location.dart';
+
+
+set urlStrategy(UrlStrategy? strategy) => customUrlStrategy = strategy;
+
+typedef PopStateListener = void Function(Object? state);
+
+/// Represents and reads route state from the browser's URL.
+///
+/// By default, the [HashUrlStrategy] subclass is used if the app doesn't
+/// specify one.
+abstract class UrlStrategy {
+  /// Abstract const constructor. This constructor enables subclasses to provide
+  /// const constructors so that they can be used in const expressions.
+  const UrlStrategy();
```

Comment:
Same here. All methods are abstract, which tells me this class can be implemented. So we don't need a constructor. The implementation can still be const.